aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -57,6 +57,54 @@ TORCH_META_FUNC(mm)(const Tensor & self, const Tensor & mat2) {
   "The input tensor must be a matrix with size ", self.sizes()[0], "x", mat2.sizes()[1], ", but got a ", result.dim(),
   "-D tensor with size ", result.sizes()[0], "x", result.sizes()[1]);
 }
+
+template <typename Meta>
+void common_checks_baddbmm_bmm(Meta& meta, const Tensor& batch1, const Tensor& batch2, const Scalar& beta, const Scalar& alpha, bool is_bmm, const Tensor& self_or_result) {
+  TORCH_CHECK(batch1.dim() == 3, "batch1 must be a 3D tensor");
+  TORCH_CHECK(batch2.dim() == 3, "batch2 must be a 3D tensor");
+
+  const auto batch1_sizes = batch1.sizes();
+  const auto batch2_sizes = batch2.sizes();
+
+  int64_t bs = batch1_sizes[0];
+  int64_t contraction_size = batch1_sizes[2];
+  int64_t res_rows = batch1_sizes[1];
+  int64_t res_cols = batch2_sizes[2];
+
+  TORCH_CHECK(batch2_sizes[0] == bs && batch2_sizes[1] == contraction_size);
+
+  if (!is_bmm) {
+    TORCH_CHECK(self_or_result.dim() == 3, "self must be a 3D tensor");
+    const auto self_sizes = self_or_result.sizes();
+    TORCH_CHECK(self_sizes[0] == bs && self_sizes[1] == res_rows && self_sizes[2] == res_cols);
+  }
+
+  namedinference::propagate_names_if_nonempty(
+    self_or_result,
+    namedinference::compute_bmm_outnames(const_cast<Tensor&>(self_or_result), batch1, batch2)
```

Comment:
Nit: do you think this `const` can also be propagated to `compute_bmm_outnames`?

Reply:
Yes, done! :)